### PR TITLE
[SYCL] Reduce cost of XPTI instrumentation

### PR DESCRIPTION
**Pull request by alexbatashev**
Argument pack added unwanted overhead. Guard it with `xptiTraceEnabled()` with the intention to guard it with new API, that would allow instrumented apps to check if the stream is observed.

### CODE REVIEW

File: sycl/source/detail/plugin.hpp [hunk 1]
```diff
@@ -175,7 +180,7 @@ class plugin {
     pi::emitFunctionEndTrace(CorrelationID, PIFnName);
     pi::emitFunctionWithArgsEndTrace(CorrelationIDWithArgs,
                                      static_cast<uint32_t>(PiApiOffset),
-                                     PIFnName, ArgsData.data(), R, *MPlugin);
+                                     PIFnName, ArgsDataPtr, R, *MPlugin);
```
Comment (smaslov-intel) [inline sycl/source/detail/plugin.hpp:183]: Should you also put lines 180-181 under if (xptiTraceEnabled())?

**Review by smaslov-intel — APPROVED** [review]
LGTM

File: sycl/source/detail/plugin.hpp (hunk 1 above)
Comment (alexbatashev) [reply sycl/source/detail/plugin.hpp:183]: I don't think so. Bodies of those functions are wrapped with `if (xptiTraceEnabled())` already. The main overhead here is from packing arguments.

Comment (smaslov-intel) [reply sycl/source/detail/plugin.hpp:183]: It just looks strange that the call to emitFunctionWithArgsBeginTrace is guarded above (I assume their bodies also have that check) but these are not. Anyways, I am fine as is if you don't want to change

Comment (alexbatashev) [reply sycl/source/detail/plugin.hpp:183]: Right, I'm planning some more changes to shut down notifications, that are not received by any party. This is a temporary fix for the problem anyway.

**Pull request merged**